src/relay/backend/interpreter.cc

```diff
@@ -486,6 +487,9 @@ class Interpreter : public ExprFunctor<ObjectRef(const Expr& n)>,
       // The output shape of shape_of must be static since Relay doesn't support
       // dynamic rank tensors.
       is_dyn = false;
+    } else if (call_node->op == ndarray_size_op_) {
```

Comment:
can you refactor the code into IsDynamic? e.g. make an IsDynamic that take a call node.